validator/degree_solver.py

```
@@ -175,10 +181,33 @@ def load_requirements(
         # Logic for adding majors
         for input_req in degree_requirements_input.majors:
             # Get major data from json
-            if input_req not in degree_plans:
+            year = degree_requirements_input.year
+            if input_req not in degree_plans[str(year)]:
+                # Check if the years before this one have it
+                y = year
+                while str(y := y - 1) in degree_plans:
+                    if input_req in degree_plans[str(y)]:
+                        year = y
+                        break
+                if (
+                    year != degree_requirements_input.year
+                ):  # The using_year has been replaced to a working year
+                    break
+                # Check if the years after this one have it
+                y = year
+                while str(y := y + 1) in degree_plans:
+                    if input_req in degree_plans[str(y)]:
+                        year = y
+                        break
+                if (
+                    year != degree_requirements_input.year
+                ):  # The using_year has been replaced to a working year
+                    break
                 print("Error: degree plan not found!")
```

Can we raise and bubble up a 404 error instead of printing here?

---

ok